pkg/udf/map_udf.go

```diff
@@ -113,10 +117,6 @@ func (u *MapUDFProcessor) Start(ctx context.Context) error {
 			// create watermark publisher using watermark stores
 			publishWatermark = jetstream.BuildPublishersFromStores(ctx, u.VertexInstance, toVertexWmStores)
 
-			readers, writers, err = buildJetStreamBufferIO(ctx, u.VertexInstance, natsClientPool)
```

Comment:
When watermark is disabled, it's not working? How come we have this bug.....

Reply:
it got introduced during a refactor

Reply:
updated one of the e2e tests to cover this case..